### ISSUE TITLE
Navigation to extensions doesn't always work

### ISSUE THREAD

**Issue by busywiththingsalreadyforgotten** [directus/directus#1966]
## Version Info
  - Directus version and branch: 607aa12 v6.4.4 (Ruby Rabbit)
  - PHP version: 5.6.31
  - MySQL version: 5.7.19
  - Web server: Apache
  - OS name and version: Windows 10

## Expected Behavior
Clicking on the extension link will navigate to the extension every time

## Actual Behavior
The first time you click on the extension link it works, subsequent attempts do not.

## Steps to Reproduce
 1.  Fresh install of directus
 2.  Rename example extension folder _extension to extension
 3.  Open directus, click on example extension link (this works)
 4. Navigate somewhere else
 5. Attempt to navigate back (nothing happens)

( I've notice that the link that appears in the status bar in chrome when you hover over the link is ext/example when you first log into directus or after a force refresh. However, after navigating to the extension once the link becomes ext/ext/example. Not sure if this hints at the underlying problem)

**Comment by spawnrider** [comment]
Same issue for me. 

I think that the issue is about the relative path 
The first navigation is **http://localhost:8080/ext/mobile**

Then switch to a table and go back to the extension ... 
The second navigation is **http://localhost:8080/tables/ext/mobile**

**Comment by rijkvanzanten** [comment]
This turns out to be a way nastier bug than I had expected.. The Router actually gets the right path when clicking the link, but the main page content view doesn't update to reflect the navigation for some reason..

AFAICT:

## Steps to Reproduce
1. Navigate to the extension
2. Navigate to a table listing and/or edit view
3. Try navigating to the extension.

## Behavior
The first navigation to the extension succeeds.
The navigation to the table / listview also succeeds.
Any subsequent request to the extension either renders an empty page or refreshes the last view (listing / edit view)

**Comment by lokers** [comment]
I would like to upvote this as I am having exactly the same problem with my custom extension and it can be reproduced in the same way as described above.

**Comment by danhipke** [comment]
I'll give this one a shot.

**Comment by danhipke** [comment]
Think I figured it out. The issue is in customs/extensions/_example/main.js

```
getMainView: function () {
      if (!this.mainView) {
        this.mainView = new MainView();
        app.router.v.main.setView('#content', this.mainView);
      }
      return this.mainView;
    },
```

The problem is that `app.router.v.main.setView('#content', this.mainView);` only gets called if there's no mainView. So if you 

1. Navigate to the extension
2. Navigate to a table listing and/or edit view
3. Try navigating to the extension.

this.mainView has already been initialized and `app.router.v.main.setView('#content', this.mainView);` doesn't get called. The fix for this is moving `app.router.v.main.setView('#content', this.mainView);` outside of the null check, like so

```
getMainView: function () {
      if (!this.mainView) {
        this.mainView = new MainView();
      }
      app.router.v.main.setView('#content', this.mainView);
      return this.mainView;
    },
```

However, I'm wondering if the extension should really be the one responsible for setting the `#content` view. Seems like the extension should only need to worry about providing the view that goes inside the `#content` view and the app should worry about making sure the `#content` view gets updated when someone navigates to the extension. @rijkvanzanten what do you think?

**Comment by rijkvanzanten** [comment]
@danhipke You're completely right! That's the way we're designing it to be in 7.0 too. Normally, I'd say we should look into this and come up with a solid solution, but since we're already actively working on a complete rewrite of all of this in Vue, lets fix it the way you proposed so we have a fix for 6.x out there 😄

**Comment by danhipke** [comment]
@rijkvanzanten Perfect. I'll create a PR with my fix.

**Comment by rijkvanzanten** [comment]
@danhipke Much appreciated :)

**Comment by danhipke** [comment]
@rijkvanzanten Created a PR (closed the one I accidentally created from the master instead of develop branch).

**Comment by spawnrider** [comment]
This is not fixing the issue. 
Do you made another settings (.htaccess, Angular routing, ...) ? 

I always have this issue : https://johndoe.com/ext/ext/mobile

**Comment by spawnrider** [comment]
My bad, that was an cache issue :(
I confirm it's working well now !

**Comment by rijkvanzanten** [comment]
I'll close this one as it has been fixed in #2075 😄 

Thanks again @danhipke !
